### PR TITLE
system monitor: add diagnostic messages

### PR DESCRIPTION
**Pull request by SammyRamone**
## Proposed changes
Adds publishing of diagnostic messages to the system monitor so that they are displayed in rqt diagnostic view.

## Related issues
partly solves https://github.com/bit-bots/bitbots_misc/issues/16

## Necessary checks
- [x] Update package version
- [x] Run `catkin build`
- [x] Write documentation
- [x] Create issues for future work
- [x] Test on your machine
- [x] Test on the robot
- [x] Put the PR on our Project board

### CODE REVIEW

**Comment by SammyRamone** [comment]
works on my machine. maybe we want other metrics to launch warnings for network problems

**Comment by SammyRamone** [comment]
needs https://github.com/bit-bots/bitbots_lowlevel/pull/68 to be launched corretly

**Comment by SammyRamone** [comment]
opened an issue for starting this also on the odroid
https://github.com/bit-bots/bitbots_misc/issues/81

**Comment by SammyRamone** [comment]
I tested it on the robot and it works. Can be merged, if you don't propose any changes for throwing warnings with the network interfaces

File: system_monitor/src/system_monitor/monitor.py [hunk 1]
```diff
@@ -39,27 +41,68 @@ def validate_single_param(param_name, required_type):
...
+        diag_cpu.name = "SYSTEMCPU"
+        diag_cpu.hardware_id = "CPU"
+        diag_mem = DiagnosticStatus()
+        diag_mem.name = "SYSTEMMemory"
+        diag_cpu.hardware_id = "Memory"
 
-            running_processes, cpu_usages = cpus.collect_all() if config['do_cpu'] else (-1, [])
+        while not rospy.is_shutdown():
+            running_processes, cpu_usages, overall_usage_percentage = cpus.collect_all() if config['do_cpu'] else (
+                -1, [])
```
Comment (timonegk) [inline system_monitor/src/system_monitor/monitor.py:61]:
```suggestion
                -1, [], 0)
```
Or some other default for `overall_usage_percentage`

**Review by timonegk — CHANGES_REQUESTED** [review]

**Review by timonegk — APPROVED** [review]

**Pull request merged**